### DISCUSSION TITLE
Display many-to-many relation

### DISCUSSION THREAD

**Discussion by alaketych** [prisma/prisma#4181] [Q&A] [answered]
I want to display many-to-many relation, but i can not understand how to make this query. For instance i have 3 models:
```javascript
model Article {
  id Int @id @default(autoincrement())
  author String
  title String
  content String
  photo String
}

model Category {
  id Int @id @default(autoincrement())
  label String
}

model ArticleCategory {
  article Article @relation(fields: [articleId], references: [id])
  articleId Int

  category Category @relation(fields: [categoryId], references: [id])
  categoryId Int

  @@id([articleId, categoryId])
}
```
So, i want to display for the component Article with name of category and Author.

**Comment by ryands17** [comment, marked as answer]
@alaketych Your relation created is incorrect. This is not a many-to-many relation as there is no connection to either `Article` or `Category`.

Many to many relations can be created in 2 ways:

1. Implicit (join table automatically managed by Prisma)

```prisma
model Article {
  id         Int        @id @default(autoincrement())
  author     String
  title      String
  content    String
  photo      String
  categories Category[]
}

model Category {
  id       Int       @id @default(autoincrement())
  label    String
  articles Article[]
}
```

To make a query for this, I would write

```ts
await prisma.article.findMany({
    select: {
      author: true,
      categories: { { select: { label: true } } },
    },
})
```

2. Explicit (with an explicit Join table)

```prisma
model Article {
  id         Int               @id @default(autoincrement())
  author     String
  title      String
  content    String
  photo      String
  categories ArticleCategory[]
}

model Category {
  id       Int               @id @default(autoincrement())
  label    String
  articles ArticleCategory[]
}

model ArticleCategory {
  article   Article @relation(fields: [articleId], references: [id])
  articleId Int

  category   Category @relation(fields: [categoryId], references: [id])
  categoryId Int

  @@id([articleId, categoryId])
}
```

To make a query for this, I would write:

```ts
await prisma.article.findMany({
    select: {
      author: true,
      categories: { select: { category: { select: { label: true } } } },
    },
})
```

You can read more about many-to-many relations [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#many-to-many-relations).

**Reply by alaketych** [reply]
Thank you! First of all reason was in incorrect models.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
